static/js/containers/DashboardPage.js

```diff
@@ -150,8 +152,12 @@ class DashboardPage extends React.Component {
 
   componentWillUnmount() {
     const { dispatch } = this.props;
+    let program = this.getCurrentlyEnrolledProgram();
     dispatch(clearDashboard(SETTINGS.user.username));
     dispatch(actions.prices.clear(SETTINGS.user.username));
+    if (program !== undefined) {
+      dispatch(actions.programLearners.clear(program.id));
+    }
```

Comment:
this will only clear for the currently enrolled program, but we could have fetch program info for, say, two programs (if the user has switched programs using the program switching widget). how important do you think it is to clear all program? something like 

```js
R.keys(programLearners.forEach(id => (
  dispatch(actions.programLearners.clear(id))
))
```

I'm not sure we need to worry about it, but idk.